### ISSUE TITLE
Flink SQL Error when trying to write a Flink changelog table to an Iceberg table

### ISSUE THREAD

**Issue by bkahloon** [apache/iceberg#2172]
I'm facing some issues with writing a Flink changelog table to an Icberg table (maybe my query is wrong). 

My Flink CDC query

```
 CREATE TABLE shipments (
  shipment_id INT,
  origin STRING,
  is_arrived BOOLEAN,
  order_date DATE
) WITH (
  'connector' = 'postgres-cdc',
  'hostname' = 'dummy-db.us-east-1.rds.amazonaws.com',
  'port' = '5432',
  'username' = 'postgres',
  'password' = '********',
  'database-name' = 'postgres',
  'schema-name' = 'public',
  'table-name' = 'shipments',
  'decoding.plugin.name'= 'pgoutput'
);
```

Creation of Iceberg table after I switch the catalog to my Iceberg catalog (catalog implementation is AWS Glue, confirmed that I'm able to connect to Glue Catalog).

```
CREATE TABLE shipments(
  shipment_id INT,
  origin STRING,
  is_arrived BOOLEAN,
  order_date DATE
)PARTITIONED BY  (order_date);

```

Then I tried to insert the changelog table into the Iceberg table

```
Flink SQL> INSERT INTO iceberg.iceberg.shipments SELECT * FROM default_catalog.default_database.shipments;
[INFO] Submitting SQL update statement to the cluster...
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.table.api.TableException: AppendStreamTableSink doesn't support consuming update and delete changes which is produced by node TableSourceScan(table=[[default_catalog, default_database, shipments]], fields=[shipment_id, origin, is_arrived, order_date])
```

**Comment by bkahloon** [comment]
@openinx could you please point me in the right direction

**Comment by openinx** [comment]
@bkahloon ,   in this iceberg 0.11.0 release,  we still don't support exporting cdc events to iceberg by using flink SQL, but data stream is supported.  Here is PR: https://github.com/apache/iceberg/commit/1b66bdfc084ac73fe999299d041aa2e5677f43c9. 

Supporting the flink SQL will be the exact goal in 0.12.0 .  For more details, you mean interested in this presentation: 
![image](https://user-images.githubusercontent.com/5028729/106230067-aa94ed00-6229-11eb-904d-ba4bca9209f3.png)

The slides share: https://docs.google.com/presentation/d/1X2KrzlAF4tsxuHHhL3OEN8EkvAb4sLdcXRxGsKFjPBY/edit#slide=id.p

**Comment by bkahloon** [comment]
Thanks a lot for the info @openinx

**Comment by bkahloon** [comment]
@openinx just a follow up question. I ingested the data using the datastream Flink CDC api. I then used the FlinkSink in Iceberg to write to the Iceberg table. However, I can't seem to figure out this behaviour.

The application reads all the rows in the source db, but then doesn't write to the Iceberg table until I cancel the job (it's as if the data gets committed and shows up in S3 once I cancel the job). I checked if there was any backpressure in the job and there was none.

From reading into the IcebergFilesCommitter, it seems that Iceberg writes the files on checkpoints ? (please correct me if I'm wrong, I didn't go through the entire implementation). I then enabled checkpoints at an interval of 30seconds and still same result. Will FlinkSink wait until it reaches the 128Mb default parquet files size in Iceberg before it writes out the file ?

**Comment by openinx** [comment]
Yes, the flink sink will commit those files for every checkpoint interval actually. If you can not see the newly committed data files in S3,  then I guess it should have some problems.   Did you recover the flink job from a savepoint ?  There's a bug that will cause the similar issue, you might want to read the issue: https://github.com/apache/iceberg/issues/2033#issue-779721770, and the current solution (before release the next flink) is here: https://github.com/apache/iceberg/issues/2033#issuecomment-769952948 

If not the case, do you have some logs that could help to dig into this problem?   Java stacktrace ?  Thanks.

**Comment by bkahloon** [comment]
@openinx Sorry, it was just an issue with how I implemented a custom debezium deserializer (wrong RowKind for some rows). After I fixed that everything works fine. 

Could maybe help me understand this scenario. Lets say in the source db we only get like 10 rows inserted in the last 1 hour and we set our Flink checkpoint interval to 1 hr. Will FlinkSink just just commit a new parquet file with these 10 rows ?
